src/common_utils/language_utils/sentence_filter_utils.py

```diff
@@ -94,4 +106,89 @@ def extract_lemmas_and_filter_stopwords(self, sentence):
         lemmas = []
         for word in words:
             lemmas.append(self.extract_lemma(word).lower())
-        return list(filter(lambda x: x is not None, lemmas))
\ No newline at end of file
+        return lemmas
+
+    def extract_complex_lemmas_and_filter_stopwords(self, phrase):
+        analysis = self.utils.morfeusz.analyse(phrase)
+        tags = set([])
+        old_word_index = 0
+        single_tag = set([])
+        #print('ANALYSIS RESULT:\n', analysis)
+        for interpretation in analysis:
+            #print("INTERPOLATION", interpretation)
+            new_word_index = interpretation[0]
+            if 'interp' == interpretation[2][2]:
+                continue
+            if new_word_index != old_word_index:
+                #     zapisz, aktualizuj index, coś jeszcze?
+                if len(single_tag) > 0:
+                    single_tags_list = list(single_tag)
+                    single_tags_list.sort()
+                    #print('single tag before adding: ', single_tag)
+                    tags.add(list_to_str_with_colons(single_tags_list))
+                old_word_index = new_word_index
+                single_tag = set([])
+            word_form = delete_additional_info_after_colon(interpretation[2][1])
+            if not self.is_stopword(word_form):
+                single_tag.add(word_form.lower())
+            #print('INTERPRETATION: ', interpretation)
+        if len(single_tag) > 0:
+            single_tags_list = list(single_tag)
+            single_tags_list.sort()
+            tags.add(list_to_str_with_colons(single_tags_list))
+        return tags
+
+    def is_stopword(self, word):
+        return word.lower() in self.stop_words
+
+    def split_to_norm_and_complex_lemmas(self, lemmas_list):
```

Comment:
this is from my branch with algorithm improvement, this is probably not used with Your changes, may be removed